### PR TITLE
add method signatures for drop-in compatibility

### LINKED ISSUE #24: FabricBlockSettings is missing some method signatures from Block.Settings
The absence of these breaks the implied drop-in replacement compatibility of the class:
* of(Material, DyeColor);
* noCollision();
* breakInstantly();
* strength(float);
* isPistonExtension();

For that matter, FabricBlockSettings has no methods at all for altering:
* private boolean pistonExtension;

### PR DESCRIPTION
**Pull request by ehwuts**
Resolves #24, hopefully with nothing extra needed to be added for pistonExtension or this might explode instead.

### CODE REVIEW

File: src/main/java/net/fabricmc/fabric/block/FabricBlockSettings.java [hunk 1]
```diff
@@ -141,6 +150,19 @@ public FabricBlockSettings strength(float hardness, float resistance) {
...
+	public FabricBlockSettings strength(float both) {
+		return strength(both, both);
+	}
+
+	public FabricBlockSettings breakInstantly() {
+		return strength(0.0F);
+	}
+
+	public FabricBlockSettings isPistonExtension() {
+		this.pistonExtension = true;
```
Comment (SquidDev) [inline src/main/java/net/fabricmc/fabric/block/FabricBlockSettings.java]: I don't think this field is defined anywhere, which means this doesn't compile.

Comment (asiekierka) [reply src/main/java/net/fabricmc/fabric/block/FabricBlockSettings.java]: Yeah, I'll be handling this myself - I just want to know where pistonExtension came from as I am of high belief that's not Mojang's intention...

...right?

Comment (asiekierka) [reply src/main/java/net/fabricmc/fabric/block/FabricBlockSettings.java]: Please do! I'm stalling this PR until then.

Comment (ehwuts) [reply src/main/java/net/fabricmc/fabric/block/FabricBlockSettings.java]: The field is declared as a property in Block and Block.Settings, and initialized by Block.Settings. I have no idea what the implications are from it making BlockState.initShapeCache do nothing, and at a glance the additional logic it activates in BaseFluid doesn't actually do anything.

..and that's the extent of my knowledge.

Comment (SquidDev) [reply src/main/java/net/fabricmc/fabric/block/FabricBlockSettings.java]: I remapped it to `hasDynamicBounds`/`dynamicBounds` in a recent PR, so think issue should be sorted now :)

**Comment by asiekierka** [comment]
Solved, with one exception (strength -> hardness) which makes a bit more sense in our context IMO.

**Comment by Prospector** [comment]
:/ would still like to keep the signature... maybe we should change the yarn mapping?

**Pull request closed without merging**